distribution/pull_v2.go

```diff
@@ -393,7 +393,7 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 	oldTagImageID, err := p.config.ReferenceStore.Get(ref)
 	if err == nil {
 		if oldTagImageID == imageID {
-			return false, nil
+			return false, addDigestReference(p.config.ReferenceStore, ref, manifestDigest, imageID)
```

Comment:
@tonistiigi: Isn't this the same as the current behavior with pull-by-digest? If you pull a manifest with digest X, you'll create a digest reference repo@X for it, but that doesn't mean you downloaded every layer and verified it as part of the pull.